### PR TITLE
CHANGELOG: Add an entry for the redesign

### PR DESCRIPTION
**Pull request by pdubroy**
Add a CHANGELOG entry for the redesign, using language from (and linking to) the [blog post](https://about.sourcegraph.com/blog/introducing-sourcegraphs-new-ui/).

CleanShot 2021-06-11 at 11 24 14

### CODE REVIEW

**Review by eseliger — APPROVED** [review]
Do we want to mention in any way that we continue to ship some more improvements soon? 
Otherwise LGTM! Thanks for doing this

File: CHANGELOG.md [hunk 1]
```diff
@@ -31,6 +31,13 @@ All notable changes to Sourcegraph are documented in this file.
...
 - Sourcegraph will now refuse to start if there are unfinished [out-of-band-migrations](https://docs.sourcegraph.com/admin/migrations) that are deprecated in the current version. See the [upgrade documentation](https://docs.sourcegraph.com/admin/updates) for changes to the upgrade process. [#20967](https://github.com/sourcegraph/sourcegraph/pull/20967)
 - Code Insight pages now have new URLs [#21856](https://github.com/sourcegraph/sourcegraph/pull/21856)
+- We are proud to bring you [an entirely new visual design for the Sourcegraph UI](https://about.sourcegraph.com/blog/introducing-sourcegraphs-new-ui/). We think you’ll find this new design improves your experience and sets the stage for some incredible features to come. Some of the highlights include:
+
+  - **Refined search results:** The redesigned search bar provides more space for expressive queries, and the new results sidebar help users discover search syntax without referencing documentation.
```
Comment (quinnkeast) [inline CHANGELOG.md:36]:
```suggestion
  - **Refined search results:** The redesigned search bar provides more space for expressive queries, and the new results sidebar helps to discover search syntax without referencing documentation.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -31,6 +31,13 @@ All notable changes to Sourcegraph are documented in this file.
...
+  - **Improved focus on code:** We’ve reduced non-essential UI elements to provide greater focus on the code itself, and positioned the most important items so they’re unobtrusive and live exactly where they are needed.
+  - **Improved layouts:** We’ve improved pages like diff diff views to make them easier to use and to help users quickly find the information they need.
```
Comment (quinnkeast) [inline CHANGELOG.md:38]:
```suggestion
  - **Improved layouts:** We’ve improved pages like diff views to make them easier to use and to help find information quickly.
```

**Comment by pdubroy** [comment]
Added a sentence about future improvements: "With bulk of the redesign complete, future releases will include more improvements along these lines."

**Review by valerybugakov — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -31,6 +31,15 @@ All notable changes to Sourcegraph are documented in this file.
...
+
+  - **Refined search results:** The redesigned search bar provides more space for expressive queries, and the new results sidebar helps to discover search syntax without referencing documentation.
+  - **Improved focus on code:** We’ve reduced non-essential UI elements to provide greater focus on the code itself, and positioned the most important items so they’re unobtrusive and live exactly where they are needed.
+  - **Improved layouts:** We’ve improved pages like diff views to make them easier to use and to help find information quickly.
+  - **New navigation:** A new global navigation provides immediate discoverability and access to current and future functionality.
+  - **Promoting extensibility:** We've brought the extension registry back to the main navigation and improved its design and navigation.
+
+  With bulk of the redesign complete, future releases will include more improvements along these lines.
```
Comment (quinnkeast) [inline CHANGELOG.md:42]:
```suggestion
  With bulk of the redesign complete, future releases will include more improvements and refinements.
```

**Review by quinnkeast — APPROVED** [review]
Just added a couple other suggestions to [remove idioms](https://about.sourcegraph.com/handbook/communication/content_guidelines/style_and_mechanics#avoid-idioms) based on the styleguide.

**Comment by pdubroy** [comment]
Thanks y'all! 🚀

**Review by eseliger — APPROVED** [review]

**Pull request merged**